### ISSUE TITLE
Token in memory not refreshed after refresh called

### ISSUE THREAD

**Issue by telou** [procore/ruby-sdk#51]
We've noticed that when an access token has expired, the token refresh is successful, but the initial API call that triggered the refresh then fails with a 401. Any made after this time work as expected. 

It looks like the in memory `token` is not refreshed after `refresh` is called, and so it still uses the old access token for that call:
```
def access_token
      token = fetch_token
      if token.expired?
        Util.log_info("Token Expired", store: store)
        refresh
      end
      token.access_token
 end
 ```

**Comment by PatKoperwas** [comment]
@njbbaer I think this was introduced in https://github.com/procore/ruby-sdk/commit/5aa7db232ab33ea2964d8f79facfe895f065b98b#, the re-assignment was removed

**Comment by njbbaer** [comment]
This has been fixed in `1.1.4`. Thanks for reporting it @telou.
